class-two-factor-core.php

```diff
@@ -150,29 +204,29 @@ public static function get_providers() {
 		/**
 		 * Filter the supplied providers.
 		 *
-		 * This lets third-parties either remove providers (such as Email), or
-		 * add their own providers (such as text message or Clef).
-		 *
 		 * @param array $providers A key-value array where the key is the class name, and
 		 *                         the value is the path to the file containing the class.
 		 */
-		$providers = apply_filters( 'two_factor_providers', $providers );
+		$additional_providers = apply_filters( 'two_factor_providers', $providers );
```

Comment:
@kasparsd Using the same existing filter seems wrong here. Shouldn't this be a new filter so you only get the additional providers?
Right now the filter is called twice in `get_providers()`, see https://github.com/WordPress/two-factor/blob/6a95e7ffdebb2e0408b4ccb298e9df16d5b6998c/class-two-factor-core.php#L264-L275.

Reply:
@kasparsd This still seems wrong to me and is now released. Why running the filter twice?

Reply:
Here is a pull request that removes the duplicate #651.